frontend/src/metabase/query_builder/components/expressions/ExpressionWidget.tsx

```diff
@@ -124,11 +124,16 @@ export const ExpressionWidget = (props: ExpressionWidgetProps): JSX.Element => {
             value={name}
             placeholder={t`Something nice and descriptive`}
             fullWidth
-            onChange={event => setName(event.target.value)}
+            onChange={event => setName(event.target.value.replace(/\s/g, ""))}
```

Comment:
The issue is there when the name consists only of whitespace characters.
Names such as `My awesome metric` must not be disallowed.

Reply:
Apologize. I've revert my changes and checked whether the name is consist of space only. I've also update the test. Thanks for your feedback! @kamilmielnik